test/jdk/java/util/Collections/Shuffle.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug     8294693
+ * @summary Basic test for Collections.shuffle
+ */
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import java.util.random.RandomGenerator;
+import java.util.random.RandomGeneratorFactory;
+
+public class Shuffle {
+    static final int N = 100;
+
+    public static void main(String[] args) {
+        test(new ArrayList<>());
+        test(new LinkedList<>());
+    }
+
+    static void test(List<Integer> list) {
+        for (int i = 0; i < N; i++) {
+            list.add(i);
+        }
+        Collections.shuffle(list);
+        if (list.size() != N) {
+            throw new RuntimeException(list.getClass() + ": size " + list.size() + " != " + N);
+        }
+        for (int i = 0; i < N; i++) {
+            if (!list.contains(i)) {
+                throw new RuntimeException(list.getClass() + ": does not contain " + i);
+            }
+        }
+        checkRandom(list);
+        checkRandomGenerator(list);
+    }
+
+    private static void checkRandomGenerator(List<Integer> list) {
+        RandomGeneratorFactory<RandomGenerator> factory = RandomGeneratorFactory.getDefault();
+        list.sort(null);
+        Collections.shuffle(list, factory.create(1));
```

Comment:
This assumes that the default factory will accept a seed value that initializes its state so that the pseudorandom sequence is repeatable. Not an unreasonable assumption, but `create(long)` essentially says that it can ignore the seed, and `getDefault` says the algorithm may change over time. On the other hand if something does change such that the pseudorandom sequence isn't repeatable, this test will most likely fail immediately. I was poking around for an explicit way to request some kind of PRNG that's guaranteed to be started in repeatable state, but I couldn't find one. Hmmm.

Reply:
Oh, that's an unfortunate detail, I didn't know about this. We can explicitly request a JumpableGenerator, like "Xoshiro256PlusPlus" and then copy it via `copy()` method. What do you think?